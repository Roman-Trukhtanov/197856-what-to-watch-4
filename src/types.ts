export type Time = number | string;

export type VideoTypes = {
  [key: string]: string;
}

export type Action = {
  [key: string]: string;
}

export type ActionObject<T> = {
  type: string;
  payload: T;
}

export type Noop = () => void;

export type FavoriteStatus = 0 | 1;

export interface Review {
  rating: number;
  comment: string;
}

export interface PreviewVideo {
  className?: string;
  src: string;
  type: string;
  isAutoPlay?: boolean;
  isLoop: boolean;
  isMute: boolean;
}
export interface FullVideo {
  className: string;
  src: string;
  type: string;
  isAutoPlay?: boolean;
  isLoop: boolean;
  isMute: boolean;
}

export interface Movie {
  id: number;
  title: string;
  genre: string;
  coverSrc: string;
  bgColor: string;
  bigPosterSrc: string;
  previewImgSrc: string;
  isFavorite: boolean;
  fullVideo: FullVideo;
  previewVideo: PreviewVideo;
  details: {
    rate: number;
    releaseYear: number;
    ratingCount: number;
    level: string;
    director: string;
    runTime: number;
    description: string;
    starring: string[];
  };
}

export interface Comment {
  id: number;
  user: {
    id: number;
    name: string;
  };
  rating: number;
  comment: string;
  date: string;
  dateText: string;
}

export interface User {
  id: number;
  name: string;
  email: string;
  avatarSrc: string;
}

export interface UserData {
  email: string;
  password: string;
}

export interface ImgBgStyle {
  display: string;
  backgroundColor: string;
}
