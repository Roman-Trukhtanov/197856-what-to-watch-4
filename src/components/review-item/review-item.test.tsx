import React from "react";
import renderer from "react-test-renderer";
import ReviewItem from "./review-item";
import {Comment} from "../../types";

const mockCommentData: Comment[] = [{
  id: 1,
  user: {
    id: 2,
    name: `Other user`,
  },
  rating: 7.2,
  comment: `Other comment`,
  date: ``,
  dateText: `June 20, 2020`,
}];

describe(`ReviewItem component`, () => {
  it(`Render ReviewItem`, () => {
    const tree = renderer
      .create(<ReviewItem
        comment={mockCommentData[0]}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
