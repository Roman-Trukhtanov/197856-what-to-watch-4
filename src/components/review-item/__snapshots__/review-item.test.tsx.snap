// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewItem component Render ReviewItem 1`] = `
<div
  className="review"
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
    >
      Other comment
    </p>
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
      >
        Other user
      </cite>
      <time
        className="review__date"
        dateTime="2016-12-20"
      >
        June 20, 2020
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  >
    7.2
  </div>
</div>
`;
